### PR TITLE
#38 - Extend eloquent tools

### LINKED ISSUE #38: Extend eloquent tools
Add handling for relations, something like:
```php
@eloquent
    Feature: Eloquent test
        Scenario: Check if model has relations
            Given there is a model User in the database
            And there is a model Post in the database
            Then the model User has many Posts
            And the model Post belongs to User

```

### PR DESCRIPTION
**Pull request by PiotrFedak**
This PR adds new functions to the eloquent trait.

- Creates a model if it does not exist as data.
- Creates the specified number of models if they are missing.
- Checking relationships between models.

This should close #38

### CODE REVIEW

File: src/Features/Traits/Eloquent.php [hunk 1]
```diff
@@ -10,17 +10,151 @@
 trait Eloquent
 {
     /**
-     * @Given there is :count :model object in database
-     * @Given there should be :count :model object in database
-     * @Given there are :count :model objects in database
-     * @Given there should be :count :model objects in database
+     * @Given there is a model :model in the database
+     * @Given there is a :model in the database
```
Comment (krzysztofrewak) [inline src/Features/Traits/Eloquent.php:14]: Maybe table with parameters that you could provide to factory's `create()` method would be handy?

File: src/Features/Traits/Eloquent.php [hunk 2]
```diff
@@ -10,17 +10,151 @@
...
+
+        Assert::assertTrue(
+            method_exists($instance, $relation),
+            "The model $model1 does not have a $relation relation method.",
+        );
+
+        $related = $instance->{$relation}();
+
+        Assert::assertInstanceOf(
+            'Illuminate\Database\Eloquent\Relations\HasOne',
```
Comment (krzysztofrewak) [inline src/Features/Traits/Eloquent.php:106]: Maybe you could move it to class with other Laravel classes?

File: src/Features/Traits/Eloquent.php [hunk 3]
```diff
@@ -10,17 +10,151 @@
...
-    public function thereAreModelsInDatabase(string $model, int $count): void
+    public function thereIsAModelInTheDatabase(string $model): void
     {
         $modelClass = $this->recognizeModelClass($model);
+
+        if (!$modelClass::query()->exists()) {
+            $modelClass::factory()->create();
+        }
+
+        Assert::assertTrue($modelClass::query()->exists(), "The model $model does not exist in the database.");
```
Comment (krzysztofrewak) [inline src/Features/Traits/Eloquent.php:24]: Oh, I don't get what's going on here.

Should this method be used to seed database or check database?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Comment by JakubKermes** [comment]
Maybe we should add creating for models without factory?

File: src/Features/Traits/Eloquent.php [hunk 4]
```diff
@@ -4,23 +4,142 @@
...
+     * @Given there is a model :model in the database
+     * @Given there is a :model in the database
      */
-    public function thereAreModelsInDatabase(string $model, int $count): void
+    public function seedModelInTheDatabase(string $model, ?TableNode $table = null): void
     {
         $modelClass = $this->recognizeModelClass($model);
+        $attributes = $table ? $table->getRowsHash() : [];
+
+        $modelClass::factory()->create($attributes);
```
Comment (JakubKermes) [inline src/Features/Traits/Eloquent.php:23]: Maybe we should add creating for models without factory?

File: src/Features/Traits/Eloquent.php [hunk 5]
```diff
@@ -4,23 +4,146 @@
...
+        $model1Class = $this->recognizeModelClass($model1);
+        $relation = Str::plural($model2);
+        $instance = $model1Class::first() ?: $model1Class::factory()->create();
+
+        Assert::assertTrue(
+            method_exists($instance, $relation),
+            "The model $model1 does not have a $relation relation method.",
+        );
+
+        LaravelRelations::assertRelation($instance, $relation, 'Illuminate\Database\Eloquent\Relations\HasMany');
```
Comment (krzysztofrewak) [inline src/Features/Traits/Eloquent.php:70]: Please look at this file: https://github.com/blumilksoftware/blt/blob/main/src/LaravelContracts.php and try to replicate its purpose here.

File: src/Features/Traits/Eloquent.php [hunk 6]
```diff
@@ -4,23 +4,146 @@
...
 use PHPUnit\Framework\Assert;
 
 trait Eloquent
 {
     /**
-     * @Given there is :count :model object in database
-     * @Given there should be :count :model object in database
-     * @Given there are :count :model objects in database
-     * @Given there should be :count :model objects in database
+     * @Given there is a model :model in the database
```
Comment (krzysztofrewak) [inline src/Features/Traits/Eloquent.php:15]: All method using tables should have `:` at the end of sentence. Here it's optional, so I would add two sentences.

File: src/Helpers/LaravelRelations.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+declare(strict_types=1);
+
+namespace Blumilk\BLT\Helpers;
+
+use PHPUnit\Framework\Assert;
+
+class LaravelRelations
+{
+    public static function assertRelation($instance, string $relation, string $relationType): void
```
Comment (krzysztofrewak) [inline src/Helpers/LaravelRelations.php:11]: Please type `$instance`. Isn't that Eloquent `Model`?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/Features/Traits/Eloquent.php [hunk 8]
```diff
@@ -4,23 +4,147 @@
...
-     * @Given there should be :count :model object in database
-     * @Given there are :count :model objects in database
-     * @Given there should be :count :model objects in database
+     * @Given there is a model :model in the database:
+     * @Given there is a :model in the database:
      */
-    public function thereAreModelsInDatabase(string $model, int $count): void
+    public function seedModelInTheDatabase(string $model, ?TableNode $table = null): void
     {
         $modelClass = $this->recognizeModelClass($model);
```
Comment (JakubKermes) [inline src/Features/Traits/Eloquent.php:21]: Maybe use RecognizeClassHelper instead
```suggestion
        $modelClass = RecognizeClassHelper::recognizeObjectClass($model);
```

**Review by krzysztofrewak — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**